include/libopencm3/stm32/common/dmamux_common_all.h

```diff
@@ -0,0 +1,214 @@
+#define DMAMUX_RGxCR_SIG_ID_MASK		0x1f
+
+/**@}*/
+
+/** @defgroup dmamux_rgsr RGSR DMA request generator interrupt status register
+@{*/
+
+/** DMAMUX_RGSR_OF Trigger OF event overrun flag */
+#define DMAMUX_RGSR_OF(rg_channel)	(1 << ((rg_channel) - 1))
+
+/**@}*/
+
+/** @defgroup dmamux_rgcfr RGCFR DMA request generator clear flag register
+@{*/
+
+/** DMAMUX_RGCFR_COF COF Clear trigger event overrun flag */
+#define DMAMUX_RGCFR_COF(rg_channel)	(1 << ((rg_channel) - 1))
+
+/**@}*/
+
+/* --- Generic values ---------------------------------------- */
+
+/** @defgroup dmamux_rg_channel DMAMUX Request Generator Channel Number
+@{*/
+#define DMAMUX_RG_CHANNEL1			1
+#define DMAMUX_RG_CHANNEL2			2
+#define DMAMUX_RG_CHANNEL3			3
+#define DMAMUX_RG_CHANNEL4			4
+/**@}*/
+
+/* --- function prototypes ------------------------------------------------- */
+
+BEGIN_DECLS
+
+void dmamux_reset_dma_channel(uint32_t dmamux, uint8_t channel);
+void dmamux_enable_dma_request_event_generation(uint32_t dmamux, uint8_t channel);
+void dmamux_disable_dma_request_event_generation(uint32_t dmamux, uint8_t channel);
+
+void dmamux_set_dma_channel_request(uint32_t dmamux, uint8_t channel, uint8_t request_id);
+uint8_t dmamux_get_dma_channel_request(uint32_t dmamux, uint8_t channel);
+
+void dmamux_enable_dma_request_sync(uint32_t dmamux, uint8_t channel);
+void dmamux_disable_dma_request_sync(uint32_t dmamux, uint8_t channel);
+
+void dmamux_set_dma_request_sync_input(uint32_t dmamux, uint8_t channel, uint8_t sync_id);
+void dmamux_set_dma_request_sync_pol(uint32_t dmamux, uint8_t channel, uint8_t polarity);
+void dmamux_set_dma_request_sync_nbreq(uint32_t dmamux, uint8_t channel, uint8_t nbreq);
+
+void dmamux_enable_dma_request_sync_overrun_interrupt(uint32_t dmamux, uint8_t channel);
+void dmamux_disable_dma_request_sync_overrun_interrupt(uint32_t dmamux, uint8_t channel);
+uint32_t dmamux_get_dma_request_sync_overrun(uint32_t dmamux, uint8_t channel);
+void dmamux_clear_dma_request_sync_overrun(uint32_t dmamux, uint8_t channel);
+
+void dmamux_reset_request_generator_channel(uint32_t dmamux, uint8_t rg_channel);
+void dmamux_enable_request_generator(uint32_t dmamux, uint8_t rg_channel);
+void dmamux_disable_request_generator(uint32_t dmamux, uint8_t rg_channel);
+
+void dmamux_set_request_generator_trigger(uint32_t dmamux, uint8_t rg_channel, uint8_t sig_id);
+void dmamux_set_request_generator_trigger_pol(uint32_t dmamux, uint8_t rg_channel, uint8_t polarity);
+void dmamux_set_request_generator_trigger_gnbreq(uint32_t dmamux, uint8_t rg_channel, uint8_t gnbreq);
+
+void dmamux_enable_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel);
+void dmamux_disable_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel);
+uint32_t dmamux_get_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel);
+void dmamux_clear_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel);
```

Comment:
wow. I mean, these are consistent namings, but wow :) getting long.

Reply:
yeah .. I'm open to any naming suggestion, or clever api refinement.